Losing connection to cluster behaves ungracefully

If you make a local `Client()`, connect the labextension to it, and then close that cluster things get unhappy

<details>

```
VM277 bokeh.min.js:575 [bokeh] Failed to connect to Bokeh server: Could not open websocket
_on_error @ VM277 bokeh.min.js:575
socket.onerror @ VM277 bokeh.min.js:575
error (async)
(anonymous) @ VM277 bokeh.min.js:575
connect @ VM277 bokeh.min.js:575
n.pull_session @ VM277 bokeh.min.js:575
(anonymous) @ VM277 bokeh.min.js:574
t.add_document_from_session @ VM277 bokeh.min.js:574
O @ VM277 bokeh.min.js:164
(anonymous) @ VM277 bokeh.min.js:164
(anonymous) @ VM277 bokeh.min.js:185
requestAnimationFrame (async)
(anonymous) @ VM277 bokeh.min.js:185
t.defer @ VM277 bokeh.min.js:185
t.embed_items @ VM277 bokeh.min.js:164
embed_document @ individual-workers:57
(anonymous) @ individual-workers:61
(anonymous) @ individual-workers:77
o.safely @ VM277 bokeh.min.js:583
fn @ individual-workers:51
VM277 bokeh.min.js:574 [bokeh] Failed to load Bokeh session wnC4rgQp9b8ixgOhjfy5cdjUaghHK7ZxBsGVM2ZcXGpb: Error: Could not open websocket
t.add_document_from_session @ VM277 bokeh.min.js:574
async function (async)
t.add_document_from_session @ VM277 bokeh.min.js:574
O @ VM277 bokeh.min.js:164
(anonymous) @ VM277 bokeh.min.js:164
(anonymous) @ VM277 bokeh.min.js:185
requestAnimationFrame (async)
(anonymous) @ VM277 bokeh.min.js:185
t.defer @ VM277 bokeh.min.js:185
t.embed_items @ VM277 bokeh.min.js:164
embed_document @ individual-workers:57
(anonymous) @ individual-workers:61
(anonymous) @ individual-workers:77
o.safely @ VM277 bokeh.min.js:583
fn @ individual-workers:51
VM277 bokeh.min.js:164 Error rendering Bokeh items: Error: Could not open websocket
    at h._on_error (VM277 bokeh.min.js:575)
    at WebSocket.socket.onerror (VM277 bokeh.min.js:575)
VM277 bokeh.min.js:575 [bokeh] Lost websocket 0 connection, 1006 ()
VM190 reconnecting-websocket.min.js:8 WebSocket connection to 'ws://127.0.0.1:8787/eventstream' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
```

</details>

![image](https://user-images.githubusercontent.com/306380/102564468-5a563700-4090-11eb-8806-bcc6160eb807.png)

--- a/src/dashboard.tsx
+++ b/src/dashboard.tsx
@@ -372,22 +372,14 @@ export class URLInput extends Widget {
           urlInfo.url,
           this._serverSettings
         );
-        // No change - valid case
-        if (result.isActive && urlInfo.isActive) {
-          return;
-        }
         // Show an error if the connection died.
         if (!result.isActive && urlInfo.isActive) {
           console.warn(
             `The connection to dask dashboard ${urlInfo.url} has been lost`
           );
         }
-        // No change - invalid case
-        if (!result.isActive && !urlInfo.isActive) {
-          return;
-        }
-        // Connection died or started
-        if (result.isActive !== urlInfo.isActive) {
+        if (!JSONExt.deepEqual(result, urlInfo)) {
+          this._urlInfo = result;
           this._urlChanged.emit({
             oldValue: urlInfo,
             newValue: result

